spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,258 @@
+from qtpy.QtWidgets import (QDialog, QHBoxLayout, QMessageBox,
+                            QLabel, QProgressBar, QPushButton, QVBoxLayout,
+                            QWidget)
+
+# Local imports
+from spyder import __version__
+from spyder.config.base import _
+from spyder.utils.icon_manager import ima
+from spyder.utils.palette import QStylePalette
+from spyder.utils.programs import is_module_installed
+
+
+# Update installation process statuses
+NO_STATUS = __version__
+DOWNLOADING_INSTALLER = _("Downloading installer")
+INSTALLING = _("Installing")
+FINISHED = _("Installation finished")
+PENDING = _("Pending update")
+CHECKING = _("Checking for updates")
+CANCELLED = _("Cancelled")
+
+
+class UpdateInstallation(QWidget):
+    """Update progress installation widget."""
+
+    def __init__(self, parent):
+        super(UpdateInstallation, self).__init__(parent)
+
+        # Left side
+        action_layout = QVBoxLayout()
+        progress_layout = QHBoxLayout()
+        self._progress_widget = QWidget(self)
+        self._progress_widget.setFixedHeight(50)
+        self._progress_bar = QProgressBar(self)
+        self._progress_bar.setFixedWidth(180)
+        self.cancel_button = QPushButton()
+        self.cancel_button.setIcon(ima.icon('DialogCloseButton'))
+        progress_layout.addWidget(self._progress_bar, alignment=Qt.AlignLeft)
+        progress_layout.addWidget(self.cancel_button)
+        self._progress_widget.setLayout(progress_layout)
+
+        self._progress_label = QLabel(_('Downloading'))
+
+        self.install_info = QLabel(
+            _("Downloading the latest Spyder version <br>"))
+
+        button_layout = QHBoxLayout()
+        self.ok_button = QPushButton(_('OK'))
+        button_layout.addStretch()
+        button_layout.addWidget(self.ok_button)
+        button_layout.addStretch()
+        action_layout.addStretch()
+        action_layout.addWidget(self._progress_label)
+        action_layout.addWidget(self._progress_widget)
+        action_layout.addWidget(self.install_info)
+        action_layout.addSpacing(10)
+        action_layout.addLayout(button_layout)
+        action_layout.addStretch()
+
+        # Layout
+        general_layout = QHBoxLayout()
+        general_layout.addLayout(action_layout)
+
+        self.setLayout(general_layout)
+
+    def update_installation_status(self, status):
+        """Update installation status (downloading, installing, finished)."""
+        self._progress_label.setText(status)
+        self.install_info.setText(status + _(" the latest version of Spyder."))
+        if status == INSTALLING:
+            self._progress_bar.setRange(0, 0)
+            self.cancel_button.hide()
+
+    def update_installation_progress(self, current_value, total):
+        """Update installation progress bar."""
+        self._progress_bar.setMaximum(total)
+        self._progress_bar.setValue(current_value)
+
+
+class UpdateInstallerDialog(QDialog):
+    """Spyder installer."""
```

Comment:
```suggestion
    """Update installer dialog."""
```

Improve this docstring a bit.